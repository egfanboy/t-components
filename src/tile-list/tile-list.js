import React from 'react';
import { withTheme } from 'styled-components';
import theme from '../utils/theme';
import { mergeTheme } from '../utils/theme/merge-theme';
import { Container, Tile, Title, Description } from './tile-list.styled';

export const TileList = withTheme(props => {
    const onTileClick = tile => () =>
        props.onTileClick && props.onTileClick(tile);

    const theme = mergeTheme(props.theme);

    if (!props.items) return null;

    return (
        <Container>
            {props.items.map(item => (
                <Tile key={item.key} theme={theme} onClick={onTileClick(item)}>
                    <Title theme={theme}>{item.title}</Title>
                    {
                        <Description theme={theme}>
                            {item.description}
                        </Description>
                    }
                </Tile>
            ))}
        </Container>
    );
});

TileList.defaultProps = {
    theme,
};
