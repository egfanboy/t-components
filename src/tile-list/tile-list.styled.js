import styled from 'styled-components';
import colors from '../utils/colors';

export const Container = styled.div`
    display: grid;

    grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
    gap: 0.5rem;
`;

export const Tile = styled.div`
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    width: 300px;
    height: 250px;
    border-radius: 6px;
    border: solid 1px
        ${({ theme }) => colors.changeOpacity(theme.grey.light, 0.3)};
    padding: 1rem;
    overflow: hidden;
    margin-bottom: 0.5rem;

    &:hover {
        cursor: pointer;
        box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
    }
`;

export const Title = styled.h2`
    color: ${({ theme }) => theme.grey.dark};
    margin: 0;
`;

export const Description = styled.p`
    color: ${({ theme }) => theme.grey.light};
    margin-top: auto;
    margin-bottom: 0;
`;
