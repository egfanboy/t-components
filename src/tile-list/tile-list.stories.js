import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import { TileList } from './tile-list';
import theme from '../utils/theme';

const Wrapper = styled.div`
    padding: 4rem;
`;

const items = [
    {
        key: 'item-1',
        title: 'SOmething1',
        description: 'I am some type of footnote',
    },
    { key: 'item-2', title: 'SOmething2', description: '' },
    { key: 'item-3', title: 'SOmething3', description: '' },
    { key: 'item-4', title: 'SOmething4', description: '' },
    { key: 'item-6', title: 'SOmething5', description: '' },
    { key: 'item-7', title: 'SOmething6', description: '' },
    { key: 'item-8', title: 'SOmething7', description: '' },
];

storiesOf('TileList', module)
    .add('Normal', () => (
        <>
            <Wrapper>
                <TileList
                    theme={theme}
                    items={items}
                    onTileClick={t => console.log({ t })}
                ></TileList>
            </Wrapper>
        </>
    ))
    .add('Empty', () => (
        <>
            <Wrapper>
                <TileList
                    theme={theme}
                    items={[]}
                    onTileClick={t => console.log({ t })}
                ></TileList>
            </Wrapper>
        </>
    ));
