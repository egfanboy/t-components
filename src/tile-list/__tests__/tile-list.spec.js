import React from 'react';

import { Tile } from '../tile-list.styled';
import { makeGetEnzymeWrapper } from '../../utils/unit-testing/get-enzyme-wrapper';
import { TileList } from '../tile-list';

const items = [
    { key: 'key1', title: 'title1', description: '' },
    { key: 'key2', title: 'title2', description: 'Some description' },
];

describe('Tile List', () => {
    const getWrapper = makeGetEnzymeWrapper(TileList);

    let wrapper;
    beforeEach(() => (wrapper = getWrapper()));
    afterEach(() => wrapper && wrapper.unmount());

    describe('rendering', () => {
        it('should match the no items snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('should match the snapshot with itemsu', () => {
            wrapper = getWrapper({ items });

            expect(wrapper).toMatchSnapshot();
        });

        it('should render one tile per item', () => {
            wrapper = getWrapper({ items });

            expect(wrapper.find(Tile).length).toEqual(items.length);
        });
    });

    describe('actions', () => {
        it('should call onFocus when the input gets focused', () => {
            const tileClick = jest.fn();

            wrapper = getWrapper({ items, onTileClick: tileClick });

            const tile = wrapper.find(Tile).first();

            tile.simulate('click', {});

            expect(tileClick).toHaveBeenCalledTimes(1);
            expect(tileClick).toHaveBeenCalledWith(items[0]);
        });
    });
});
