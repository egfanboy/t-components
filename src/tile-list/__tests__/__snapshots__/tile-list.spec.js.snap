// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile List rendering should match the no items snapshot 1`] = `
<WithTheme(Component)
  theme={
    Object {
      "error": String {
        "0": "#",
        "1": "d",
        "2": "3",
        "3": "2",
        "4": "f",
        "5": "2",
        "6": "f",
        "dark": "#9e0707",
        "light": "#e54242",
      },
      "grey": String {
        "0": "#",
        "1": "7",
        "2": "3",
        "3": "7",
        "4": "3",
        "5": "7",
        "6": "3",
        "dark": "#202020",
        "light": "#838383",
      },
      "primary": String {
        "0": "#",
        "1": "5",
        "2": "7",
        "3": "0",
        "4": "7",
        "5": "e",
        "6": "b",
        "dark": "#4a0072",
        "light": "#6217ed",
      },
      "success": String {
        "0": "#",
        "1": "3",
        "2": "0",
        "3": "a",
        "4": "6",
        "5": "4",
        "6": "a",
        "dark": " #178b30",
        "light": "#42b45b",
      },
      "warning": String {
        "0": "#",
        "1": "f",
        "2": "f",
        "3": "9",
        "4": "9",
        "5": "2",
        "6": "6",
        "dark": "#ff8800",
        "light": "#ffa726",
      },
    }
  }
>
  <Component
    theme={
      Object {
        "error": String {
          "0": "#",
          "1": "d",
          "2": "3",
          "3": "2",
          "4": "f",
          "5": "2",
          "6": "f",
          "dark": "#9e0707",
          "light": "#e54242",
        },
        "grey": String {
          "0": "#",
          "1": "7",
          "2": "3",
          "3": "7",
          "4": "3",
          "5": "7",
          "6": "3",
          "dark": "#202020",
          "light": "#838383",
        },
        "primary": String {
          "0": "#",
          "1": "5",
          "2": "7",
          "3": "0",
          "4": "7",
          "5": "e",
          "6": "b",
          "dark": "#4a0072",
          "light": "#6217ed",
        },
        "success": String {
          "0": "#",
          "1": "3",
          "2": "0",
          "3": "a",
          "4": "6",
          "5": "4",
          "6": "a",
          "dark": " #178b30",
          "light": "#42b45b",
        },
        "warning": String {
          "0": "#",
          "1": "f",
          "2": "f",
          "3": "9",
          "4": "9",
          "5": "2",
          "6": "6",
          "dark": "#ff8800",
          "light": "#ffa726",
        },
      }
    }
  />
</WithTheme(Component)>
`;

exports[`Tile List rendering should match the snapshot with itemsu 1`] = `
<WithTheme(Component)
  items={
    Array [
      Object {
        "description": "",
        "key": "key1",
        "title": "title1",
      },
      Object {
        "description": "Some description",
        "key": "key2",
        "title": "title2",
      },
    ]
  }
  theme={
    Object {
      "error": String {
        "0": "#",
        "1": "d",
        "2": "3",
        "3": "2",
        "4": "f",
        "5": "2",
        "6": "f",
        "dark": "#9e0707",
        "light": "#e54242",
      },
      "grey": String {
        "0": "#",
        "1": "7",
        "2": "3",
        "3": "7",
        "4": "3",
        "5": "7",
        "6": "3",
        "dark": "#202020",
        "light": "#838383",
      },
      "primary": String {
        "0": "#",
        "1": "5",
        "2": "7",
        "3": "0",
        "4": "7",
        "5": "e",
        "6": "b",
        "dark": "#4a0072",
        "light": "#6217ed",
      },
      "success": String {
        "0": "#",
        "1": "3",
        "2": "0",
        "3": "a",
        "4": "6",
        "5": "4",
        "6": "a",
        "dark": " #178b30",
        "light": "#42b45b",
      },
      "warning": String {
        "0": "#",
        "1": "f",
        "2": "f",
        "3": "9",
        "4": "9",
        "5": "2",
        "6": "6",
        "dark": "#ff8800",
        "light": "#ffa726",
      },
    }
  }
>
  <Component
    items={
      Array [
        Object {
          "description": "",
          "key": "key1",
          "title": "title1",
        },
        Object {
          "description": "Some description",
          "key": "key2",
          "title": "title2",
        },
      ]
    }
    theme={
      Object {
        "error": String {
          "0": "#",
          "1": "d",
          "2": "3",
          "3": "2",
          "4": "f",
          "5": "2",
          "6": "f",
          "dark": "#9e0707",
          "light": "#e54242",
        },
        "grey": String {
          "0": "#",
          "1": "7",
          "2": "3",
          "3": "7",
          "4": "3",
          "5": "7",
          "6": "3",
          "dark": "#202020",
          "light": "#838383",
        },
        "primary": String {
          "0": "#",
          "1": "5",
          "2": "7",
          "3": "0",
          "4": "7",
          "5": "e",
          "6": "b",
          "dark": "#4a0072",
          "light": "#6217ed",
        },
        "success": String {
          "0": "#",
          "1": "3",
          "2": "0",
          "3": "a",
          "4": "6",
          "5": "4",
          "6": "a",
          "dark": " #178b30",
          "light": "#42b45b",
        },
        "warning": String {
          "0": "#",
          "1": "f",
          "2": "f",
          "3": "9",
          "4": "9",
          "5": "2",
          "6": "6",
          "dark": "#ff8800",
          "light": "#ffa726",
        },
      }
    }
  >
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "ceyuyJ",
              "rules": Array [
                "
    display: grid;

    grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
    gap: 0.5rem;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-bdVaJa ceyuyJ"
        >
          <styled.div
            key="key1"
            onClick={[Function]}
            theme={
              Object {
                "error": String {
                  "0": "#",
                  "1": "d",
                  "2": "3",
                  "3": "2",
                  "4": "f",
                  "5": "2",
                  "6": "f",
                  "dark": "#9e0707",
                  "light": "#e54242",
                },
                "grey": String {
                  "0": "#",
                  "1": "7",
                  "2": "3",
                  "3": "7",
                  "4": "3",
                  "5": "7",
                  "6": "3",
                  "dark": "#202020",
                  "light": "#838383",
                },
                "primary": String {
                  "0": "#",
                  "1": "5",
                  "2": "7",
                  "3": "0",
                  "4": "7",
                  "5": "e",
                  "6": "b",
                  "dark": "#4a0072",
                  "light": "#6217ed",
                },
                "success": String {
                  "0": "#",
                  "1": "3",
                  "2": "0",
                  "3": "a",
                  "4": "6",
                  "5": "4",
                  "6": "a",
                  "dark": " #178b30",
                  "light": "#42b45b",
                },
                "warning": String {
                  "0": "#",
                  "1": "f",
                  "2": "f",
                  "3": "9",
                  "4": "9",
                  "5": "2",
                  "6": "6",
                  "dark": "#ff8800",
                  "light": "#ffa726",
                },
              }
            }
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "fWCcxo",
                    "rules": Array [
                      "
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    width: 300px;
    height: 250px;
    border-radius: 6px;
    border: solid 1px
        ",
                      [Function],
                      ";
    padding: 1rem;
    overflow: hidden;
    margin-bottom: 0.5rem;

    &:hover {
        cursor: pointer;
        box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
    }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
              theme={
                Object {
                  "error": String {
                    "0": "#",
                    "1": "d",
                    "2": "3",
                    "3": "2",
                    "4": "f",
                    "5": "2",
                    "6": "f",
                    "dark": "#9e0707",
                    "light": "#e54242",
                  },
                  "grey": String {
                    "0": "#",
                    "1": "7",
                    "2": "3",
                    "3": "7",
                    "4": "3",
                    "5": "7",
                    "6": "3",
                    "dark": "#202020",
                    "light": "#838383",
                  },
                  "primary": String {
                    "0": "#",
                    "1": "5",
                    "2": "7",
                    "3": "0",
                    "4": "7",
                    "5": "e",
                    "6": "b",
                    "dark": "#4a0072",
                    "light": "#6217ed",
                  },
                  "success": String {
                    "0": "#",
                    "1": "3",
                    "2": "0",
                    "3": "a",
                    "4": "6",
                    "5": "4",
                    "6": "a",
                    "dark": " #178b30",
                    "light": "#42b45b",
                  },
                  "warning": String {
                    "0": "#",
                    "1": "f",
                    "2": "f",
                    "3": "9",
                    "4": "9",
                    "5": "2",
                    "6": "6",
                    "dark": "#ff8800",
                    "light": "#ffa726",
                  },
                }
              }
            >
              <div
                className="sc-bwzfXH fWCcxo"
                onClick={[Function]}
              >
                <styled.h2
                  theme={
                    Object {
                      "error": String {
                        "0": "#",
                        "1": "d",
                        "2": "3",
                        "3": "2",
                        "4": "f",
                        "5": "2",
                        "6": "f",
                        "dark": "#9e0707",
                        "light": "#e54242",
                      },
                      "grey": String {
                        "0": "#",
                        "1": "7",
                        "2": "3",
                        "3": "7",
                        "4": "3",
                        "5": "7",
                        "6": "3",
                        "dark": "#202020",
                        "light": "#838383",
                      },
                      "primary": String {
                        "0": "#",
                        "1": "5",
                        "2": "7",
                        "3": "0",
                        "4": "7",
                        "5": "e",
                        "6": "b",
                        "dark": "#4a0072",
                        "light": "#6217ed",
                      },
                      "success": String {
                        "0": "#",
                        "1": "3",
                        "2": "0",
                        "3": "a",
                        "4": "6",
                        "5": "4",
                        "6": "a",
                        "dark": " #178b30",
                        "light": "#42b45b",
                      },
                      "warning": String {
                        "0": "#",
                        "1": "f",
                        "2": "f",
                        "3": "9",
                        "4": "9",
                        "5": "2",
                        "6": "6",
                        "dark": "#ff8800",
                        "light": "#ffa726",
                      },
                    }
                  }
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "gxnPpf",
                          "rules": Array [
                            "
    color: ",
                            [Function],
                            ";
    margin: 0;
",
                          ],
                        },
                        "displayName": "styled.h2",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "h2",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    theme={
                      Object {
                        "error": String {
                          "0": "#",
                          "1": "d",
                          "2": "3",
                          "3": "2",
                          "4": "f",
                          "5": "2",
                          "6": "f",
                          "dark": "#9e0707",
                          "light": "#e54242",
                        },
                        "grey": String {
                          "0": "#",
                          "1": "7",
                          "2": "3",
                          "3": "7",
                          "4": "3",
                          "5": "7",
                          "6": "3",
                          "dark": "#202020",
                          "light": "#838383",
                        },
                        "primary": String {
                          "0": "#",
                          "1": "5",
                          "2": "7",
                          "3": "0",
                          "4": "7",
                          "5": "e",
                          "6": "b",
                          "dark": "#4a0072",
                          "light": "#6217ed",
                        },
                        "success": String {
                          "0": "#",
                          "1": "3",
                          "2": "0",
                          "3": "a",
                          "4": "6",
                          "5": "4",
                          "6": "a",
                          "dark": " #178b30",
                          "light": "#42b45b",
                        },
                        "warning": String {
                          "0": "#",
                          "1": "f",
                          "2": "f",
                          "3": "9",
                          "4": "9",
                          "5": "2",
                          "6": "6",
                          "dark": "#ff8800",
                          "light": "#ffa726",
                        },
                      }
                    }
                  >
                    <h2
                      className="sc-htpNat gxnPpf"
                    >
                      title1
                    </h2>
                  </StyledComponent>
                </styled.h2>
                <styled.p
                  theme={
                    Object {
                      "error": String {
                        "0": "#",
                        "1": "d",
                        "2": "3",
                        "3": "2",
                        "4": "f",
                        "5": "2",
                        "6": "f",
                        "dark": "#9e0707",
                        "light": "#e54242",
                      },
                      "grey": String {
                        "0": "#",
                        "1": "7",
                        "2": "3",
                        "3": "7",
                        "4": "3",
                        "5": "7",
                        "6": "3",
                        "dark": "#202020",
                        "light": "#838383",
                      },
                      "primary": String {
                        "0": "#",
                        "1": "5",
                        "2": "7",
                        "3": "0",
                        "4": "7",
                        "5": "e",
                        "6": "b",
                        "dark": "#4a0072",
                        "light": "#6217ed",
                      },
                      "success": String {
                        "0": "#",
                        "1": "3",
                        "2": "0",
                        "3": "a",
                        "4": "6",
                        "5": "4",
                        "6": "a",
                        "dark": " #178b30",
                        "light": "#42b45b",
                      },
                      "warning": String {
                        "0": "#",
                        "1": "f",
                        "2": "f",
                        "3": "9",
                        "4": "9",
                        "5": "2",
                        "6": "6",
                        "dark": "#ff8800",
                        "light": "#ffa726",
                      },
                    }
                  }
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": false,
                          "lastClassName": "fjxobY",
                          "rules": Array [
                            "
    color: ",
                            [Function],
                            ";
    margin-top: auto;
    margin-bottom: 0;
",
                          ],
                        },
                        "displayName": "styled.p",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "p",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    theme={
                      Object {
                        "error": String {
                          "0": "#",
                          "1": "d",
                          "2": "3",
                          "3": "2",
                          "4": "f",
                          "5": "2",
                          "6": "f",
                          "dark": "#9e0707",
                          "light": "#e54242",
                        },
                        "grey": String {
                          "0": "#",
                          "1": "7",
                          "2": "3",
                          "3": "7",
                          "4": "3",
                          "5": "7",
                          "6": "3",
                          "dark": "#202020",
                          "light": "#838383",
                        },
                        "primary": String {
                          "0": "#",
                          "1": "5",
                          "2": "7",
                          "3": "0",
                          "4": "7",
                          "5": "e",
                          "6": "b",
                          "dark": "#4a0072",
                          "light": "#6217ed",
                        },
                        "success": String {
                          "0": "#",
                          "1": "3",
                          "2": "0",
                          "3": "a",
                          "4": "6",
                          "5": "4",
                          "6": "a",
                          "dark": " #178b30",
                          "light": "#42b45b",
                        },
                        "warning": String {
                          "0": "#",
                          "1": "f",
                          "2": "f",
                          "3": "9",
                          "4": "9",
                          "5": "2",
                          "6": "6",
                          "dark": "#ff8800",
                          "light": "#ffa726",
                        },
                      }
                    }
                  >
                    <p
                      className="sc-bxivhb fjxobY"
                    />
                  </StyledComponent>
                </styled.p>
              </div>
            </StyledComponent>
          </styled.div>
          <styled.div
            key="key2"
            onClick={[Function]}
            theme={
              Object {
                "error": String {
                  "0": "#",
                  "1": "d",
                  "2": "3",
                  "3": "2",
                  "4": "f",
                  "5": "2",
                  "6": "f",
                  "dark": "#9e0707",
                  "light": "#e54242",
                },
                "grey": String {
                  "0": "#",
                  "1": "7",
                  "2": "3",
                  "3": "7",
                  "4": "3",
                  "5": "7",
                  "6": "3",
                  "dark": "#202020",
                  "light": "#838383",
                },
                "primary": String {
                  "0": "#",
                  "1": "5",
                  "2": "7",
                  "3": "0",
                  "4": "7",
                  "5": "e",
                  "6": "b",
                  "dark": "#4a0072",
                  "light": "#6217ed",
                },
                "success": String {
                  "0": "#",
                  "1": "3",
                  "2": "0",
                  "3": "a",
                  "4": "6",
                  "5": "4",
                  "6": "a",
                  "dark": " #178b30",
                  "light": "#42b45b",
                },
                "warning": String {
                  "0": "#",
                  "1": "f",
                  "2": "f",
                  "3": "9",
                  "4": "9",
                  "5": "2",
                  "6": "6",
                  "dark": "#ff8800",
                  "light": "#ffa726",
                },
              }
            }
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "fWCcxo",
                    "rules": Array [
                      "
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    width: 300px;
    height: 250px;
    border-radius: 6px;
    border: solid 1px
        ",
                      [Function],
                      ";
    padding: 1rem;
    overflow: hidden;
    margin-bottom: 0.5rem;

    &:hover {
        cursor: pointer;
        box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
    }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
              theme={
                Object {
                  "error": String {
                    "0": "#",
                    "1": "d",
                    "2": "3",
                    "3": "2",
                    "4": "f",
                    "5": "2",
                    "6": "f",
                    "dark": "#9e0707",
                    "light": "#e54242",
                  },
                  "grey": String {
                    "0": "#",
                    "1": "7",
                    "2": "3",
                    "3": "7",
                    "4": "3",
                    "5": "7",
                    "6": "3",
                    "dark": "#202020",
                    "light": "#838383",
                  },
                  "primary": String {
                    "0": "#",
                    "1": "5",
                    "2": "7",
                    "3": "0",
                    "4": "7",
                    "5": "e",
                    "6": "b",
                    "dark": "#4a0072",
                    "light": "#6217ed",
                  },
                  "success": String {
                    "0": "#",
                    "1": "3",
                    "2": "0",
                    "3": "a",
                    "4": "6",
                    "5": "4",
                    "6": "a",
                    "dark": " #178b30",
                    "light": "#42b45b",
                  },
                  "warning": String {
                    "0": "#",
                    "1": "f",
                    "2": "f",
                    "3": "9",
                    "4": "9",
                    "5": "2",
                    "6": "6",
                    "dark": "#ff8800",
                    "light": "#ffa726",
                  },
                }
              }
            >
              <div
                className="sc-bwzfXH fWCcxo"
                onClick={[Function]}
              >
                <styled.h2
                  theme={
                    Object {
                      "error": String {
                        "0": "#",
                        "1": "d",
                        "2": "3",
                        "3": "2",
                        "4": "f",
                        "5": "2",
                        "6": "f",
                        "dark": "#9e0707",
                        "light": "#e54242",
                      },
                      "grey": String {
                        "0": "#",
                        "1": "7",
                        "2": "3",
                        "3": "7",
                        "4": "3",
                        "5": "7",
                        "6": "3",
                        "dark": "#202020",
                        "light": "#838383",
                      },
                      "primary": String {
                        "0": "#",
                        "1": "5",
                        "2": "7",
                        "3": "0",
                        "4": "7",
                        "5": "e",
                        "6": "b",
                        "dark": "#4a0072",
                        "light": "#6217ed",
                      },
                      "success": String {
                        "0": "#",
                        "1": "3",
                        "2": "0",
                        "3": "a",
                        "4": "6",
                        "5": "4",
                        "6": "a",
                        "dark": " #178b30",
                        "light": "#42b45b",
                      },
                      "warning": String {
                        "0": "#",
                        "1": "f",
                        "2": "f",
                        "3": "9",
                        "4": "9",
                        "5": "2",
                        "6": "6",
                        "dark": "#ff8800",
                        "light": "#ffa726",
                      },
                    }
                  }
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "gxnPpf",
                          "rules": Array [
                            "
    color: ",
                            [Function],
                            ";
    margin: 0;
",
                          ],
                        },
                        "displayName": "styled.h2",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "h2",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    theme={
                      Object {
                        "error": String {
                          "0": "#",
                          "1": "d",
                          "2": "3",
                          "3": "2",
                          "4": "f",
                          "5": "2",
                          "6": "f",
                          "dark": "#9e0707",
                          "light": "#e54242",
                        },
                        "grey": String {
                          "0": "#",
                          "1": "7",
                          "2": "3",
                          "3": "7",
                          "4": "3",
                          "5": "7",
                          "6": "3",
                          "dark": "#202020",
                          "light": "#838383",
                        },
                        "primary": String {
                          "0": "#",
                          "1": "5",
                          "2": "7",
                          "3": "0",
                          "4": "7",
                          "5": "e",
                          "6": "b",
                          "dark": "#4a0072",
                          "light": "#6217ed",
                        },
                        "success": String {
                          "0": "#",
                          "1": "3",
                          "2": "0",
                          "3": "a",
                          "4": "6",
                          "5": "4",
                          "6": "a",
                          "dark": " #178b30",
                          "light": "#42b45b",
                        },
                        "warning": String {
                          "0": "#",
                          "1": "f",
                          "2": "f",
                          "3": "9",
                          "4": "9",
                          "5": "2",
                          "6": "6",
                          "dark": "#ff8800",
                          "light": "#ffa726",
                        },
                      }
                    }
                  >
                    <h2
                      className="sc-htpNat gxnPpf"
                    >
                      title2
                    </h2>
                  </StyledComponent>
                </styled.h2>
                <styled.p
                  theme={
                    Object {
                      "error": String {
                        "0": "#",
                        "1": "d",
                        "2": "3",
                        "3": "2",
                        "4": "f",
                        "5": "2",
                        "6": "f",
                        "dark": "#9e0707",
                        "light": "#e54242",
                      },
                      "grey": String {
                        "0": "#",
                        "1": "7",
                        "2": "3",
                        "3": "7",
                        "4": "3",
                        "5": "7",
                        "6": "3",
                        "dark": "#202020",
                        "light": "#838383",
                      },
                      "primary": String {
                        "0": "#",
                        "1": "5",
                        "2": "7",
                        "3": "0",
                        "4": "7",
                        "5": "e",
                        "6": "b",
                        "dark": "#4a0072",
                        "light": "#6217ed",
                      },
                      "success": String {
                        "0": "#",
                        "1": "3",
                        "2": "0",
                        "3": "a",
                        "4": "6",
                        "5": "4",
                        "6": "a",
                        "dark": " #178b30",
                        "light": "#42b45b",
                      },
                      "warning": String {
                        "0": "#",
                        "1": "f",
                        "2": "f",
                        "3": "9",
                        "4": "9",
                        "5": "2",
                        "6": "6",
                        "dark": "#ff8800",
                        "light": "#ffa726",
                      },
                    }
                  }
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": false,
                          "lastClassName": "fjxobY",
                          "rules": Array [
                            "
    color: ",
                            [Function],
                            ";
    margin-top: auto;
    margin-bottom: 0;
",
                          ],
                        },
                        "displayName": "styled.p",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "p",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    theme={
                      Object {
                        "error": String {
                          "0": "#",
                          "1": "d",
                          "2": "3",
                          "3": "2",
                          "4": "f",
                          "5": "2",
                          "6": "f",
                          "dark": "#9e0707",
                          "light": "#e54242",
                        },
                        "grey": String {
                          "0": "#",
                          "1": "7",
                          "2": "3",
                          "3": "7",
                          "4": "3",
                          "5": "7",
                          "6": "3",
                          "dark": "#202020",
                          "light": "#838383",
                        },
                        "primary": String {
                          "0": "#",
                          "1": "5",
                          "2": "7",
                          "3": "0",
                          "4": "7",
                          "5": "e",
                          "6": "b",
                          "dark": "#4a0072",
                          "light": "#6217ed",
                        },
                        "success": String {
                          "0": "#",
                          "1": "3",
                          "2": "0",
                          "3": "a",
                          "4": "6",
                          "5": "4",
                          "6": "a",
                          "dark": " #178b30",
                          "light": "#42b45b",
                        },
                        "warning": String {
                          "0": "#",
                          "1": "f",
                          "2": "f",
                          "3": "9",
                          "4": "9",
                          "5": "2",
                          "6": "6",
                          "dark": "#ff8800",
                          "light": "#ffa726",
                        },
                      }
                    }
                  >
                    <p
                      className="sc-bxivhb fjxobY"
                    >
                      Some description
                    </p>
                  </StyledComponent>
                </styled.p>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </Component>
</WithTheme(Component)>
`;
